state.go

```diff
@@ -317,6 +309,54 @@ func (m *Memberlist) probeNode(node *nodeState) {
 	m.suspectNode(&s)
 }
 
+// Ping initiates a ping to the node with the specified name.
+func (m *Memberlist) Ping(node string, addr net.IP, port uint16) (*time.Duration, error) {
```

Comment:
It feels a little weird to have a pointer return for the RTT, can you change that to a regular value and just return 0 in the error cases?